pkg/controller/util.go

```diff
@@ -83,7 +84,7 @@ func markAsDetached(client kubernetes.Interface, va *storage.VolumeAttachment) (
 	if err != nil {
 		return va, err
 	}
-	newVA, err := client.StorageV1beta1().VolumeAttachments().Patch(va.Name, types.MergePatchType, patch)
+	newVA, err := client.StorageV1().VolumeAttachments().Patch(va.Name, types.MergePatchType, patch, "status")
```

Comment:
if finalizers  changed,  it seems that we can not patch the changed finalize success if we only patch `status`.

if we should split `patch finalizers` and patch `status` into 2 steps?

Reply:
Yes, status and non-status fields must be patched separately